pkg/plugin/plugin.go

```diff
@@ -0,0 +1,136 @@
+package plugin
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Info struct {
+	Name             string          `json:"display_name"`
+	ConfigAttributes json.RawMessage `json:"config_attrs"`
+}
+
+type Contact struct {
+	FullName  string     `json:"full_name"`
+	Addresses []*Address `json:"addresses"`
+}
+
+type Address struct {
+	Type    string `json:"type"`
+	Address string `json:"address"`
+}
+
+type Incident struct {
+	Id                int64  `json:"id"`
+	ObjectDisplayName string `json:"object_display_name"`
+}
+
+type Event struct {
+	Time     time.Time `json:"time"`
+	URL      string    `json:"url"`
+	Type     string    `json:"type"`
+	Severity string    `json:"severity"`
+	Username string    `json:"username"`
+	Message  string    `json:"message"`
+}
+
+type NotificationRequest struct {
+	Contact       *Contact `json:"contact"`
+	Incident      Incident `json:"incident"`
+	Event         *Event   `json:"event"`
+	IcingaWeb2Url string   `json:"icingaweb2_url"`
+}
+
+type JsonRpcRequest struct {
+	Method string          `json:"method"`
+	Params json.RawMessage `json:"params"`
+	Id     uint64          `json:"id"`
+}
+
+type JsonRpcResponse struct {
+	Result json.RawMessage `json:"result"`
+	Error  string          `json:"error"`
+	Id     uint64          `json:"id"`
+}
+
+type Plugin interface {
+	Send(req *NotificationRequest) error
+	LoadConfig(jsonStr string) error
```

Comment:
You can just use `json.RawMessage` as type for that parameter. Otherwise, you're just casting it around anyways.